pkg/asset/cluster/tfvars.go

```diff
@@ -199,7 +199,7 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		for i, m := range workers {
 			workerConfigs[i] = m.Spec.Template.Spec.ProviderSpec.Value.Object.(*awsprovider.AWSMachineProviderConfig)
 		}
-		data, err := awstfvars.TFVars(vpc, privateSubnets, publicSubnets, installConfig.Config.Publish, masterConfigs, workerConfigs)
+		data, err := awstfvars.TFVars(vpc, privateSubnets, publicSubnets, installConfig.Config.AWS.PublicZoneID, installConfig.Config.Publish, masterConfigs, workerConfigs)
```

Comment:
what is the install config doesn't have the zone id set by the user??

Reply:
Fixed.